xc/xc7/yosys/synth.tcl

```diff
@@ -33,7 +33,38 @@ if { [info exists ::env(INPUT_XDC_FILE)] && $::env(INPUT_XDC_FILE) != "" } {
 write_verilog $::env(OUT_SYNTH_V).premap.v
 
 # Map Xilinx tech library to 7-series VPR tech library.
-read_verilog -lib $::env(TECHMAP_PATH)/cells_sim.v
+read_verilog -specify -lib $::env(TECHMAP_PATH)/cells_sim.v
+
+# Convert congested CARRY4 outputs to LUTs.
+techmap -map  $::env(TECHMAP_PATH)/carry_map.v
+write_json $::env(OUT_JSON).carry_fixup.json
+exec $::env(PYTHON3) $::env(TECHMAP_PATH)/fix_carry.py < $::env(OUT_JSON).carry_fixup.json > $::env(OUT_JSON).carry_fixup_out.json
+design -push
+read_json $::env(OUT_JSON).carry_fixup_out.json
+
+techmap -map  $::env(TECHMAP_PATH)/clean_carry_map.v
+
+# Re-read baseline libraries
+read_verilog -lib -specify -D_EXPLICIT_CARRY +/xilinx/cells_sim.v
```

Comment:
Why is EXPLICIT_CARRY needed here if we are transitioning to have the native CARRY map from yosys?

Reply:
It's a mistake

Reply:
Fixed